code/modules/crafting/slapcrafting/_crafting_stage.dm

```diff
@@ -20,6 +20,28 @@
 		stages += GET_DECL(nid)
 	next_stages = stages
 
+/decl/crafting_stage/proc/generate_completion_string()
+	var/list/names = assemble_name_strings()
+	if(ispath(completion_trigger_type, /obj/item/stack) || stack_consume_amount)
+		names.Insert(1, max(stack_consume_amount, 1))
+	return jointext(names, " ")
+
+/decl/crafting_stage/proc/assemble_name_strings()
+	SHOULD_CALL_PARENT(TRUE)
+	var/list/names = list()
+	var/obj/item/prop = completion_trigger_type
+	if(ispath(prop, /obj/item/stack))
```

Comment:
occurs to me that stack.get_string_for_amount is an instance proc which is unpleasant, otherwise it could be used here. that STATIC_CALL macro I wrote might work but it's also scary